src/content/docs/en/reference/routing-reference.mdx

```diff
@@ -0,0 +1,312 @@
+---
+title: Routing Reference
+i18nReady: true
+tableOfContents:
+  minHeadingLevel: 2
+  maxHeadingLevel: 4
+---
+import Since from '~/components/Since.astro';
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro';
+import ReadMore from '~/components/ReadMore.astro';
+
+There is no separate routing configuration in Astro.
+
+Every [supported page file](/en/basics/astro-pages/#supported-page-files) located within the special `src/pages/` directory creates a route. When the file name contains a [parameter](#params), a route can create multiple pages dynamically, otherwise it creates a single page.
+
+By default, all Astro page routes and endpoints are generated and prerendered at build time. [On-demand server rendering](/en/guides/on-demand-rendering/) can be set for individual routes, or as the default.
+
+## `prerender`
+
+<p>
+
+**Type:** `boolean`<br />
+**Default:** `true` <br />
```

Comment:
I already approved, but with Chris suggestions (🙌🏽 ) the content has changed a bit. Everything looks fine with a quick read (I didn't check for typo), but I'm a bit concern about this `Default:`.

Both `Per-page override` and `Switch to server mode` are under this `h2`. And while `true` is the default in static mode, this is not the case (from my understanding) with `server` mode. So it could be confusing.
I think we should either move `Default:` into subsections or use a slightly more complex `Default:`. Perhaps something like:
```suggestion
**Default:** `true` with `output: 'static'` or `false` with `output: 'server'` <br />
```

Reply:
Thanks for commenting on this! I had wondered about this, because you're right, it's not the default in `server` mode. It *is* true that it's default **until you do something to change the nature of your project**, which is how the text was handling it. I wasn't sure whether we had a standard way of either/or 'ing it, or whether it was in fact actually helpful to present the default as `true` (because it is until something else happens.)

Reply:
I was hesitant about that. I understood the intention given the structure of the page, but I am now quite familiar with how it works (I think 😄 ). So I totally understand what you mean.
But for someone new it might be less obvious... especially if they quickly scan the page instead of reading the explanations in detail...
(When I see some questions asked, sometimes I just want to say "but have you read the doc"!? I admire your patience! 🙄) So I don't know what the best way to solve this is, but I think it could be problematic.

Regarding the standard way of doing it, I don't think so. But I'm sure there is another place where `Default:` is used as I suggested (two possible values, not the wording). But I can't find the page anymore... I don't think it's been edited, maybe moved. Or I'm not looking in the right place.

Reply:
I had thought we had another place too, but I didn't immediately find it and didn't want to introduce inconsistency! If you can't find it either, maybe it didn't make it into the new docs!

I'm happy to go with your suggestion if you feel it's more helpful!

Reply:
Ok, so I only just saw your comment now... (thanks the email notifications...) I was on the page, but it seems I'm having auto-refresh issues with Github the last few days (notifs in particular). Even after posting the link, your comment wasn't displayed... 

> I'm happy to go with your suggestion if you feel it's more helpful! My only slight nagging concern is that `output: 'static'` is a default that no one would set. And, it's def true by default default, even if they don't know what `output: 'static'` is. Like, this is _really_ the default. And I _do_ think there's value in it being presented that way.

Yeah, I understand your concern and honestly I don't have a perfect solution. Considering the way the content is presented, I would say it is fine the way it is. But then I think that some people read the doc too quickly and they risk stopping at the word in bold (`Default:`) without reading the rest and that could cause confusion.

So I thought I'd just mention it just in case. Now, especially since we're not sure what's best, I also agree to leave it as is, and if it's a problem, we'll update later.

Reply:
Thank you for the conversation! I'm leaning towards leaving it like this and then waiting to see whether it causes confusion or more people mention it. It can always be changed!

I think part of why I have this opinion is the reason behind making the old hybrid behaviour the default is because too many people were simply enabling `output: 'server'` just to have ANY on-demand rendering features, even when they only needed it on one page. We're also trying to position the docs to nudge towards the behaviour of thinking of your site as prerendered, opting in per-route when you need it, and only when and if you realize you *really* need it, switching to `server` mode. I do think this framing helps that story as well!